test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -0,0 +1,386 @@
+    addAllPlayerUnits();
+    final IntegerMap<PlayerID> playerUnitCounts = uc.getPlayerUnitCounts();
+    assertThat(playerUnitCounts.getInt(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(playerUnitCounts.getInt(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
+  }
+
+  @Test
+  public void hasUnitsFromMultiplePlayers() {
+    addAllDefaultPlayerUnits();
+    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(false)));
+    addAllOtherPlayerUnits();
+    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(true)));
+  }
+
+  @Test
+  public void getHolder() {
+    assertThat(uc.getHolder(), is(equalTo(defaultPlayerID)));
+  }
+
+  @Test
+  public void allMatch() {
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(false)));
+  }
+
+  @Test
+  public void someMatch() {
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeTwo)), is(equalTo(false)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeTwo)), is(equalTo(true)));
+  }
+
+  @Test
+  public void countMatches() {
+    addAllDefaultPlayerUnits();
+    assertThat(uc.countMatches(Matches.unitIsOfType(unitTypeOne)), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(uc.countMatches(Matches.unitIsOfType(unitTypeTwo)), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+  }
+
+  @Test
+  public void getMatches() {
+    addAllDefaultPlayerUnits();
+    assertThat(uc.getMatches(Matches.unitIsOfType(unitTypeOne)).size(), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(uc.getMatches(Matches.unitIsOfType(unitTypeTwo)).size(), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+    addAllOtherPlayerUnits();
+    assertThat(uc.getMatches(Matches.unitIsOfType(unitTypeOne)).size(), is(equalTo(unitCountUnitTypeOne)));
+    assertThat(uc.getMatches(Matches.unitIsOfType(unitTypeTwo)).size(), is(equalTo(unitCountUnitTypeTwo)));
+  }
+
+  @Test
+  public void iterator() {
+    final Collection<Unit> unitsOfDefaultPlayerOfUnitTypeTwo = getUnitsOfDefaultPlayerOfUnitTypeTwo();
+    uc.addAllUnits(unitsOfDefaultPlayerOfUnitTypeTwo);
+    final Iterator<Unit> ucIterator = uc.iterator();
+    final Iterator<Unit> collectionIterator = unitsOfDefaultPlayerOfUnitTypeTwo.iterator();
+    while (ucIterator.hasNext()) {
+      final Unit ucIteratorUnit = ucIterator.next();
+      final Unit collectionIteratorIteratorUnit = collectionIterator.next();
+      assertThat(ucIteratorUnit, is(equalTo(collectionIteratorIteratorUnit)));

```

Comment:
`assertThat(ucIteratorUnit, is(equalTo(collectionIteratorIteratorUnit)));`
Can simplify to:
`assertThat(unIterator.next(), is(collectionIterator.next()));